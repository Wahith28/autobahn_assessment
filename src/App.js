import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import "./App.css";
import Navbar from "./components/Navbar";

const App = () => {
  const dispatch = useDispatch();
  const selector = useSelector((state) => state);
  useEffect(() => {
    fetchData();
  }, []);

  const fetchData = async () => {
    await fetch("https://jsonplaceholder.typicode.com/users")
      .then((res) => res.json())
      .then((data) => {
        dispatch({ type: "GET_DATA", payload: data });
      })
      .catch((err) => {
        console.log(err);
      });
  };

  const onAdd = async (name, email) => {
    await fetch("https://jsonplaceholder.typicode.com/users", {
      method: "POST",
      body: JSON.stringify({
        name: name,
        email: email
      }),
      headers: {
        "Content-type": "application/json; charset=UTF-8"
      }
    })
      .then((res) => {
        if (res.status !== 201) {
          return;
        } else {
          return res.json();
        }
      })
      .then((data) => {
        dispatch({ type: "ADD_USER", payload: [...selector, data] });
      })
      .catch((err) => {
        console.log(err);
      });
  };

  const onEdit = async (input, id) => {
    await fetch(`https://jsonplaceholder.typicode.com/users/${id}`, {
      method: "PUT",
      body: JSON.stringify({
        input
      }),
      headers: {
        "Content-type": "application/json; charset=UTF-8"
      }
    })
      .then((res) => {
        dispatch({ type: "EDIT_USER", payload: input });
      })
      .catch((err) => {
        console.log(err);
      });
  };

  const onDelete = async (id) => {
    await fetch(`https://jsonplaceholder.typicode.com/users/${id}`, {
      method: "DELETE"
    })
      .then((res) => {
        const temp = selector.filter((user) => {
          return user.id !== id;
        });
        dispatch({ type: "DELETE_USER", payload: temp });
      })
      .catch((err) => {
        console.log(err);
      });
  };
  return (
    <div className="App">
      <Navbar
        onAdd={onAdd}
        users={selector}
        onDelete={onDelete}
        onEdit={onEdit}
        data={selector}
      />
      <br />
    </div>
  );
};

export default App;
