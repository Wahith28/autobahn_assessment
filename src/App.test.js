
import { Provider } from "react-redux";
import React from "react";
import { shallow } from "enzyme";
import App from "./App";
import configureStore from "redux-mock-store";

describe("AppComponent", () => {
  const initialState = { output: 10 };
  const mockStore = configureStore();
  let store;
  it('should render correctly', () => {
    store = mockStore(initialState);
    const component = shallow(
      <Provider store={store}>
        <App />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
