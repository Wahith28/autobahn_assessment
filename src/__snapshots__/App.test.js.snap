// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should render correctly 1`] = `ShallowWrapper {}`;
